### PR TITLE
Add a yapf style config file.

### LINKED ISSUE #2428: cuda-quantum needs a .style.yapf file added at the top-level directory
Generating this file should be easy with yapf itself. Adding the file to the source tree will prevent having to search for the "proper" yaof command line settings and prevent accidentally reformatting python code to "something weird".

### PR DESCRIPTION
**Pull request by schweitzpgi**
This makes the default "google" style and will help prevent people from accidentally using the yapf default style.

Fix #2428

### CODE REVIEW

**Review by bmhowe23 — COMMENTED** [review]
FWIW, we put this into CUDA-QX. Hopefully it is equivalent to what you have here?

https://github.com/NVIDIA/cudaqx/blob/main/.style.yapf

**Comment by schweitzpgi** [comment]
> FWIW, we put this into CUDA-QX. Hopefully it is equivalent to what you have here?
> 
> https://github.com/NVIDIA/cudaqx/blob/main/.style.yapf

Yes. The file being added is the output from the tool itself for the "google" style. I guess the verbose format is intended to make sure any changes to default settings in future releases of yapf don't cause all the source code to get reformatted? That's a "known" issue when using clang-format, for example.

**Comment by bmhowe23** [comment]
> I guess the verbose format is intended to make sure any changes to default settings in future releases of yapf don't cause all the source code to get reformatted? That's a "known" issue when using clang-format, for example.

I suspect we should consider updating .github/workflows/repo_checks.yml, too, because it is calling `yapf --style google`, which would override this style file. That is - if a future release of yapf changed the default settings, that part of our CI would be unhappy with people attempting to use the yapf style file in the PR.

**Review by khalatepradnya — APPROVED** [review]
LGTM

File: .style.yapf [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[style]
+based_on_style = google
```
Comment (khalatepradnya) [inline .style.yapf:2]: Seeing the nature of this file, I assume it does not need a copyright banner?

Comment (schweitzpgi) [reply .style.yapf:2]: I think the IP is firmly at 0%. heh

**Review by bmhowe23 — APPROVED** [review]

**Pull request merged**